code/WorkInProgress/MadmanMartian/blacksmithing/misc_components.dm

```diff
@@ -15,7 +15,13 @@
 	if(is_type_in_list(I, finishing_requirements))
 		to_chat(user, "<span class = 'notice'>You begin to attach \the [I] to \the [src].</span>")
 		if(do_after(user, src, 4 SECONDS))
-			user.drop_item(I)
+			if(istype(I, /obj/item/stack))
+				var/obj/item/stack/S = I
+				if(!S.use(1))
+					return
+			else
+				if(!user.drop_item(I))
+					return
 			finishing_requirements.Remove(I.type)
 			gen_quality(quality-I.quality, quality, I.material_type)
 			qdel(I)
```

Comment:
This qdel will also delete the whole stack after you've used one piece, won't it? Also to note, use() will do ReturnToPool if uses up the last in the stack.

Reply:
good catch